dashboard/lti_new.py

```diff
@@ -0,0 +1,198 @@
+from django.views.decorators.csrf import csrf_exempt
+from pylti1p3.contrib.django import DjangoOIDCLogin, DjangoMessageLaunch, DjangoCacheDataStorage
+from pylti1p3.tool_config import ToolConfDict
+from django.contrib.auth.models import User
+from dashboard.common.db_util import canvas_id_to_incremented_id
+from django.core.exceptions import ObjectDoesNotExist
+from dashboard.models import Course, CourseViewOption
+from django.conf import settings
+
+
+logger = logging.getLogger(__name__)
+INSTRUCTOR = 'http://purl.imsglobal.org/vocab/lis/v2/membership#Instructor'
+TA = 'http://purl.imsglobal.org/vocab/lis/v2/membership/Instructor#TeachingAssistant'
+COURSE_MEMBERSHIP = 'membership'
+
+
+def get_tool_conf():
+    lti_config = settings.LTI_CONFIG
+    try:
+        tool_config = ToolConfDict(lti_config)
+    except Exception as e:
+        raise e
+    # There should be one key per platform and the name relay on platforms generic domain not institution specific
+    platform_domain = list(lti_config.keys())[0]
+    client_id = lti_config[platform_domain][0]['client_id']
+    private_key_file_path = lti_config[platform_domain][0]['private_key_file']
+    public_key_file_path = lti_config[platform_domain][0]['public_key_file']
+    public_key = public_key_file_path if public_key_file_path else '/secret/public.key'
+    private_key = private_key_file_path if private_key_file_path else '/secret/private.key'
+    try:
+        private_key_content = open(private_key, 'r').read()
+        public_key_content = open(public_key, 'r').read()
+    except OSError as e:
+        raise e
+    tool_config.set_public_key(platform_domain, public_key_content, client_id=client_id)
+    tool_config.set_private_key(platform_domain, private_key_content, client_id=client_id)
+    return tool_config
+
+
+def get_jwks(request):
+    try:
+        tool_conf = get_tool_conf()
+    except (OSError, Exception) as e:
+        error_message = {
+            'lti_error_message': f'Fetching JWKS failed due to {e}'
+        }
+        return JsonResponse(error_message, status=500)
+    return JsonResponse(tool_conf.get_jwks())
+
+
+def check_if_instructor(roles, username, course_id):
+    if INSTRUCTOR in roles:
+        logger.info(f'user {username} is Instructor in the course {course_id}')
+        return True
+    return False
+
+
+def course_user_roles(roles, username):
+    user_roles_in_course = [role for role in roles if COURSE_MEMBERSHIP in role]
+    if len(user_roles_in_course) == 0:
+        logger.info(f'User {username} do not have course membership role, must be admin {roles}')
+        return list()
+    short_role_str_list = set()
+    for role in user_roles_in_course:
+        short_role_str_list.add(role)
+    if INSTRUCTOR in short_role_str_list and TA in short_role_str_list:
+        logger.info(f'User {username} is a {TA} in the course')
+        short_role_str_list.remove(INSTRUCTOR)
+    return list(short_role_str_list)
```

Comment:
Since you've added the full role URIs and you no longer need to split the roles on "`#`", this can now be simplified to:

```suggestion
    user_membership_roles = set([role for role in roles if COURSE_MEMBERSHIP in role])
    if not user_membership_roles:
        logger.info(f'User {username} does not have course membership roles, must be admin {roles}')
    elif INSTRUCTOR in user_membership_roles and TA in user_membership_roles:
        logger.info(f'User {username} is a {TA} in the course')
        user_membership_roles.remove(INSTRUCTOR)
    return user_membership_roles
```